src/resources/UsageAnalytics/Read/DataHealth/DataHealthInterfaces.ts

```diff
@@ -0,0 +1,291 @@
+    /**
+     * Optional set of visit ids to filter events on.
+     * Entries are OR-ed together, if other filters are specified they are combined using AND.
+     */
+    visitId?: string[] | string;
+    /**
+     * A set of event sources for which the events will be returned.
+     * If not specified, events are not filtered on event sources.
+     */
+    eventSource?: string[] | string;
+    /**
+     * A set of event types for which the events will be returned.
+     * If not specified, events are not filtered on event types.
+     */
+    eventType?: string[] | string;
+    /**
+     * A set of search hubs for which the events will be returned.
+     * If not specified, events are not filtered on search hub.
+     */
+    searchHub?: string[] | string;
+    /**
+     * A health status for which the events will be returned.
+     * If not specified, events are not filtered on event types.
+     */
+    healthStatus?: string;
+    trackingId?: string[];
+}
+
+export interface DataHealthListEventsParameters
+    extends OrganizationParamParts,
+        TimeRangeParamParts,
+        DataHealthFilterEventsParamParts,
+        Paginated {}
+
+export interface DataHealthGetEventPayloadParams extends OrganizationParamParts {
+    /**
+     * The ID of the event to get the payload off.
+     */
+    eventId: string;
+    /**
+     * The ISO-8601 formatted timestamp of the event.
+     */
+    timestamp: string;
+}
+
+export interface DataHealthEventPayloadResponse {
+    /**
+     * The ID of the event.
+     */
+    eventId: string;
+    /**
+     * The timestamp of the event.
+     */
+    timestamp: string;
+    /**
+     * The raw payload of the event.
+     */
+    eventPayload: Record<string, unknown>;
+}
+
+export interface DataHealthValidationErrorModel {
+    /**
+     * The error code.
+     */
+    code: string;
+    /**
+     * The path of the parameter causing the error.
+     */
+    path: string;
+    /**
+     * A description of the error.
+     */
+    message: string;
+}
+
+export interface DataHealthEventsListItemModel {
+    /**
+     * The source of the event.
+     */
+    source: string;
+    /**
+     * The type of the event.
+     */
+    type: string;
+    /**
+     * The searchHub of the event.
+     */
+    searchHub: string;
+    /**
+     * The ISO-8601 formatted timestamp of the event.
+     */
+    timestamp: string;
+    /**
+     * The visitId of the event.
+     */
+    visitId: string;
+    /**
+     * The clientId of the event.
+     */
+    clientId: string;
+    /**
+     * The id of the event.
+     */
+    eventId: string;
+    /**
+     * The validation errors of the event.
+     */
+    validationErrors: DataHealthValidationErrorModel[];
+}
+
+export interface DataHealthListEventsResponse extends PaginatedResponse {
+    /**
+     * The listing of events in the current page.
+     */
+    events: DataHealthEventsListItemModel[];
+}
+export interface DataHealthListFacetValueParams
```

Comment:
Very minor detail: there are empty newlines between other exports and type definitions, you could do that here too (for consistency).
```suggestion

export interface DataHealthListFacetValueParams
```